### ISSUE TITLE
Download Complete NOTICE doesn't report correct saved file path

### ISSUE THREAD

**Issue by AndCycle** [aria2/aria2#635]
here is a sample metalink4 file,

```
<?xml version="1.0" ?>
<metalink xmlns="urn:ietf:params:xml:ns:metalink">
	<file name="dir/becreated/file.saved">
		<url>http://localhost/place/to/download/file</url>
	</file>
</metalink>
```

as you can see the file name imply file will be saved into certain directory structure,
and it do saved to that path relatively,

> D:\test\dir\becreated\file.saved

but the download complete notice tells another story,
which is kind of confuse.

> 04/21 19:44:47 [NOTICE] Download complete: D:/test/file.saved

related part

> 
> RequestGroup::reportDownloadFinished
> https://github.com/aria2/aria2/blob/c9e2223741f9627f27b3d2befef48acdd79d504d/src/RequestGroup.cc#L1150

**Comment by tatsuhiro-t** [comment]
Thank you for telling us.  We'll look into it.

**Comment by tatsuhiro-t** [comment]
It works for me at least on Linux

```
04/23 17:45:42 [NOTICE] Download complete: /downloads/dir/becreated/file.saved
```

**Comment by tatsuhiro-t** [comment]
It also works for me on Windows.

**Comment by tatsuhiro-t** [comment]
Are you using aria2 latest version?

**Comment by AndCycle** [comment]
current version, I think I provide a invalid sample for this, 
because I spot this symptoms on a custom made meta4 file which contain thousands of file,
which is not suitable for a bug report,
I will refine a correct sample on this.

**Comment by AndCycle** [comment]
sorry for the invalid sample, here is a working one,
which you can spot there is an extra download complete notice

```
<?xml version="1.0" ?>
<metalink xmlns="urn:ietf:params:xml:ns:metalink">
	<file name="bat/eng.bat/ch1-1.bat">
		<url>https://s3-ap-northeast-1.amazonaws.com/allstars-personal/content/class/bat/eng.bat/ch1-1.bat</url>
		<metaurl mediatype="torrent">https://s3-ap-northeast-1.amazonaws.com/allstars-personal/content/class/bat/eng.bat/ch1-1.bat?torrent</metaurl>
	</file>
	<file name="bat/eng.bat/ch2-1.bat">
		<url>https://s3-ap-northeast-1.amazonaws.com/allstars-personal/content/class/bat/eng.bat/ch2-1.bat</url>
		<metaurl mediatype="torrent">https://s3-ap-northeast-1.amazonaws.com/allstars-personal/content/class/bat/eng.bat/ch2-1.bat?torrent</metaurl>
	</file>
</metalink>
```
 
> D:\test>aria2c.exe --seed-time=0 test-b.meta4
> 
> 04/23 17:20:36 [WARN] Gaining privilege SeManageVolumePrivilege failed.
> 
> 04/23 17:20:36 [WARN] --file-allocation=falloc will not work properly.
> 
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/ch2-1.bat
> 
> 04/23 17:20:37 [NOTICE] IPv4 DHT: listening on UDP port 6963
> 
> 04/23 17:20:37 [NOTICE] IPv4 BitTorrent: listening on TCP port 6947
> 
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/ch1-1.bat
> 
> 04/23 17:20:37 [NOTICE] Seeding is over.
> 
> 04/23 17:20:37 [NOTICE] Seeding is over.
> 
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/bat/eng.bat/ch1-1.bat
> 
> 04/23 17:20:37 [NOTICE] Your share ratio was 0.0, uploaded/downloaded=0B/135B
> 
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/bat/eng.bat/ch2-1.bat
> 
> 04/23 17:20:37 [NOTICE] Your share ratio was 0.0, uploaded/downloaded=0B/135B
> 
> Download Results:
> gid   |stat|avg speed  |path/URI
> ======+====+===========+=======================================================
> 601672|OK  |   1.2KiB/s|D:/test/bat/eng.bat/ch1-1.bat
> 06db07|OK  |     849B/s|D:/test/bat/eng.bat/ch2-1.bat
> 
> Status Legend:
> (OK):download completed.

**Comment by tatsuhiro-t** [comment]
What is the expected result compared to what you actually get?

**Comment by AndCycle** [comment]
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/ch2-1.bat
> 04/23 17:20:37 [NOTICE] Download complete: D:/test/ch1-1.bat

just wondering about this two line, why they are there?
does the file actually save there once then moved to the destination?
if it is so is there any filename conflict issue?

**Comment by tatsuhiro-t** [comment]
Thank you.  They came from the download of torrent files (e.g., https://s3-ap-northeast-1.amazonaws.com/allstars-personal/content/class/bat/eng.bat/ch1-1.bat?torrent).  But it is a bug that it shows the actual path since they are processed in memory.
Fix committed via 3c637fa

**Comment by AndCycle** [comment]
thanks for the fix
